rpc.go

```diff
@@ -248,14 +248,23 @@ func GetCollectionInfo(r *hfile.Reader, keySampleSize int) (*gen.HFileInfo, erro
 		for next && found < keySampleSize {
 			if rand.Float64() < pr {
 				buf[found] = it.Key()
+				found++
 			}
 			next, err = it.Next()
 			if err != nil {
 				return nil, err
 			}
 		}
+		lastFound := found-1
```

Comment:
Yes. Without it you end up with a bunch of nil keys at the end. The scala
code rewinds the cursor, which is not great either. I might switch to using
reservoir sampling.

On Mon, Oct 5, 2015 at 7:58 AM, David Taylor <notifications@github.com>
wrote:

> In rpc.go
> <https://github.com/foursquare/quiver/pull/6#discussion_r41153473>:
>
> >  			}
> >  			next, err = it.Next()
> >  			if err != nil {
> >  				return nil, err
> >  			}
> >  		}
> > +		lastFound := found-1
>
> this pads out the sample by repeating the last one?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/foursquare/quiver/pull/6/files#r41153473>.
>